autolab/Autolab
# Lint views/submissions

## Summary

### Summary generated by Reviewpad on 18 Sep 23 15:50 UTC
This pull request includes multiple changes across different files:

1. The code_viewer.html.erb file has several modifications, including removal of unnecessary whitespace, modifications to link_to code, addition of onclick event, and changes to div tags and classes.
2. The missing.html.erb file has changes related to link_to methods, including updates to paths and titles. There are also structural changes in loops, table rows, and conditional statements.
3. The destroyConfirm.html.erb file has minor changes in line formatting, header modification, and class modification.
4. The app/views/submissions/_annotation_pane.html.erb file has changes related to span class modification, link_to tag formatting, and curly braces in the link_to method options.
5. The edit.html.erb file has changes including the addition of a parameter, adjustments in indentation and spacing, modifications of link_to method, and adjustments in form fields.
6. The app/views/submissions/view.html.erb file has changes related to title variable, stylesheet links, JavaScript code, and HTML tags, including the addition of attributes, styles, and a prerender attribute.
7. The viewPDF.html.erb file has changes related to the assignment of @title, addition of CSS and external stylesheet links, styling for specific elements, JavaScript tags, JSON assignments, and modifications of HTML tags.
8. There are changes in the form.html.erb file related to path updates, rearrangement of form fields, updates in input fields, inclusion of hidden fields, and modifications of select options and submit button.
9. The annotationsHelp.html.erb file has been deleted, containing instructions and examples for annotating code.
10. The _file_tree.html.erb file has changes related to indentation and modifications of link_to method.
11. The app/views/submissions/view.js.erb file has changes related to modifications of variables including curly braces and removal of a specific line.
12. The app/views/submissions/_version_links.html.erb file has changes related to the update of link paths for previous and next versions of a submission.
13. Updates in URLs for downloading PDF files in the _annotations_js.html.erb view file and removal of an empty line are noted.
14. Changes in the code regarding syntax and function improvement in the registerComponent method are highlighted.
15. The app/views/submissions/_annotation_form.html.erb file has changes related to placeholder attribute, closing tag of input element, removal of id attribute, addition of autocomplete attribute, and modifications of title attribute.
16. The changes in the app/views/submissions/_grades.html.erb file include adding a newline at the end of the file.
17. The app/views/submissions/_code_symbol_tree.html.erb file has changes related to added whitespace and adjusted indentation in the code block.
18. The app/views/submissions/_version_dropdown.html.erb file has changes related to added line breaks, indentation adjustments, comments, modified value attribute, and a space in the method.
19. The app/views/submissions/index.html.erb has changes including removal of trailing white spaces, indentation fixes, replacements of deprecated routes, and adjustments in link formatting.

Please let me know if you need more information or assistance with anything else.


## Description

- Lint `views/submissions`, using `bundle exec erblint app/views/*(folder)/*(file).html.erb -a`
- Replace bracket links with function (_path()) links
- removed `annotationsHelp.html.erb` because it did not seem to be used anymore (annotation docs are in a separate location anyways)
- Change some forms (Edit submission) to use `FormBuilderWithDateTimeInput` to have standardized form layout

## Motivation and Context


- As part of summer linting

## How Has This Been Tested?



- Verify "Manage Submissions" renders, links work to edit, destory, etc
- Verify "Edit Submission" works, "Destroy Submission" works
- Verify viewing a submission and code, annotations, downloading, etc. all work

## Checklist:
